pkg/binding/controller.go

```diff
@@ -267,6 +280,23 @@ func (c *Controller) run(ctx context.Context, workers int) error {
 	return nil
 }
 
+func (c *Controller) createManagedClustersInformer(ctx context.Context) {
+	informerFactory := ocm.GetOCMInformerFactory(c.ocmClientset)
+	informer := informerFactory.Cluster().V1().ManagedClusters().Informer()
+	_, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
In this specific case, one function for all events Add/Update/Delete is very inefficient.
with smarter logic and pre checks, you can avoid the big overhead of re-evaluation of all placements.